packages/edit-post/src/store/selectors.js

```diff
@@ -236,7 +254,40 @@ export function getActiveMetaBoxLocations( state ) {
  * @return {boolean} Whether the meta box location is active.
  */
 export function isMetaBoxLocationActive( state, location ) {
-	return getActiveMetaBoxLocations( state ).includes( location );
+	return getMetaBoxesPerLocation( state, location ).length !== 0;
+}
+
+/**
+ * Returns the list of all the available meta boxes for a given location.
+ *
+ * @param {Object} state    Global application state.
+ * @param {string} location Meta box location to test.
+ *
+ * @return {Array} List of meta boxes.
+ */
+export function getMetaBoxesPerLocation( state, location ) {
+	return get(
+		state.metaBoxes.locations,
+		[ location ],
+		[]
```

Comment:
The default value will have a different reference on every call.

We probably should have some a reusable const or helper which always returns the same instance of an empty array or object to avoid re-renders on every call.